libs/@local/hql/diagnostics/examples/jexpr.rs

```diff
@@ -0,0 +1,78 @@
+use core::fmt::Debug;
+use std::io::stdout;
+
+use hql_diagnostics::{
+    category::Category, config::ReportConfig, help::Help, label::Label, severity::Severity,
+    span::DiagnosticSpan, Diagnostic,
+};
+use hql_span::{storage::SpanStorage, Span, SpanId, TextRange};
+use jsonptr::PointerBuf;
+
+#[derive(Debug, Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+struct JsonSpan {
+    range: TextRange,
+    pointer: Option<PointerBuf>,
+    parent_id: Option<SpanId>,
+}
+
+impl Span for JsonSpan {
+    fn parent_id(&self) -> Option<SpanId> {
+        self.parent_id
+    }
+}
+
+// The error is at the `-`
+const SOURCE: &str = r#"["let", "x-y", {"const": 2}]"#;
```

Comment:
Can we change the example to `["let", "x-y", {"const": 2}]`? Or is it too complicated due to the inner type?

Reply:
considering not doing this, because it is just too complicated for a report example